util/find-doc-nits

```diff
@@ -203,6 +205,153 @@ sub check_head_style {
+            return $unwrapped;
+        } else {
+            return "";          # Malformed option
+        }
+    }
+    return undef;               # Something else
+}
+
+sub option_check {
+    my ($id, $filename, $contents) = @_;
+    my $synopsis = ($contents =~ /=head1\s+SYNOPSIS(.*?)=head1/s, $1);
+
+    # Some pages have more than one OPTIONS section, let's make sure
+    # to get them all
+    my $options = '';
+    while($contents =~ /=head1\s+[A-Z ]*?OPTIONS$(.*?)(?==head1)/msg) {
+        $options .= $1;
+    }
+
+    my %synopsis_opts = ();
+    my %options_opts = ();
+
+
+    # Look for options with no or incorrect markup
+    while ($synopsis =~
+           /(?<![-<[:alnum:]])-(?:$markup_re|.)*(?![->[:alnum:]])/msg) {
+        err($id, "Malformed option [1] in SYNOPSIS: $&");
+    }
+
+    while ($synopsis =~ /$markup_re/msg) {
+        my $found = $&;
+        print STDERR "$id:DEBUG[option_check] SYNOPSIS: found $found\n"
+            if $debug;
+        my $option_uw = is_option_normalise($id, $filename, $found);
+        if ($option_uw) {
+            $synopsis_opts{$option_uw} = 1 if defined $option_uw;
+        } elsif (defined $option_uw) {
+            err($id, "Malformed option [2] in SYNOPSIS: $found");
+        }
+    }
+
+    # In OPTIONS, we look for =item paragraphs.
+    # (?=^\s*$) detects an empty line.
+    while ($options =~ /=item\s+(.*?)(?=^\s*$)/msg) {
+        my $item = $&;
+
+        while ($item =~ /(\[\s*)?($markup_re)/msg) {
+            my $found = $2;
+            print STDERR "$id:DEBUG[option_check] OPTIONS: found $&\n"
+                if $debug;
+            err($id, "Unexpected bracket in OPTIONS =item: $item")
+                if ($1 // '') ne '' && $found =~ /^B<\s*-/;
+
+            my $option_uw = is_option_normalise($id, $filename, $found);
+            if ($option_uw) {
+                $options_opts{$option_uw} = 1 if defined $option_uw;
+            } elsif (defined $option_uw) {
+                err($id, "Malformed option in OPTIONS: $found");
+            }
+        }
+    }
+
+    foreach (sort keys %synopsis_opts) {
+        err($id, "Option $_ from SYNOPSIS not present in OPTIONS")
+            unless defined $options_opts{$_};
+    }
+
+    foreach (sort keys %options_opts) {
+        err($id, "Option $_ from OPTIONS not present in SYNOPSIS")
+            unless defined $synopsis_opts{$_};
+    }
+}
+
+# Normal symbol form
+my $symbol_re = qr/[[:alpha:]_][_[:alnum:]]*?/;
+
+# Checks of function name (man3) formatting.  The man3 checks are
+# easier than the man1 checks, we only check the names followed by (),
+# and only the names that have POD markup.
+
+sub functionname_check {
+    my ($id, $filename, $contents) = @_;
+
+    while ($contents =~ /($markup_re)\(\)/msg) {
+        print STDERR "$id:DEBUG[functionname_check] SYNOPSIS: found $&\n"
+            if $debug;
+
+        my $symbol = $1;
+        (my $unmarked = $symbol) =~ s/[BIL]<|>//msg;
```

Comment:
haven't used that nested assignment before, would prefer two lines for consistency within the file.

Reply:
I'm pretty sure you can find similar ones in other scripts. I do that pretty often, and like your dislike of else efter return, I have a dislike for the multi-line variant of this...

Reply:
i know TMTOWTTD, but consistency within a single source file doesn't seem like too much to ask for.

Reply:
Okie

Reply:
(the alternative is butt ugly, but fine...)